### PR TITLE
Cache eleventy_img generated images

### PR DESCRIPTION
**Pull request by zeroby0**
Adds filesystem caching of images generated by eleventy_img to speed up build time. Cache is persisted locally and on Netlify.

Case | Before | After
---|---|---
Build time | 48 seconds | 12 seconds
Serve time | 42 seconds | 7 seconds

Serve time is the time from running `yarn run serve` and content being available on `localhost:4001`.

Read more at:
- https://github.com/zeroby0/netlify-plugin-11ty/
- https://github.com/11ty/eleventy-img/pull/116

If this is merged, I'll notify here when 116 is merged. All the images look okay in local testing, but [please test when merging](https://deploy-preview-99--youthful-wilson-d44ff5.netlify.app/).

Thank you for opensourcing your website, it was very helpful in verifying that the patch is error-less. Here is a PR as a token of gratitude :)

### CODE REVIEW

**Comment by AleksandrHovhannisyan** [comment]
This is awesome, thanks again! I'll test this thoroughly, but I may hold off on merging until your PR to the Eleventy image plugin gets reviewed, just to be on the safe side.

**Comment by zeroby0** [comment]
Glad you like it! :D

I hope it gets reviewed soon. It's a pretty simple change though, it's puzzling how no one has reviewed it yet. I should probably make a tool to make the review easier.

File: netlify.toml
Comment (AleksandrHovhannisyan) [inline netlify.toml:12]: Does setting `cache_img_httpHeader` to `false` disable caching for images, or is this for something else?

**Review by AleksandrHovhannisyan — COMMENTED** [review]
Just one question! Gonna also test this locally when I get a chance; the Netlify deploy preview looks good 😄

**Review by AleksandrHovhannisyan — APPROVED** [review]
Looks good locally!

I didn't notice a change in local build times; it's definitely faster on Netlify, though.

**Comment by zeroby0** [comment]
Even on rebuilds?
The first local build would take just as much time as normal, but subsequent rebuilds should be much faster.
Either that or your computer is really really fast and I'm jealous 😄

**Comment by AleksandrHovhannisyan** [comment]
Yeah, rebuild took around 90 seconds total (which is typical for this site locally). My laptop is likely slower than yours, lol.

**Comment by zeroby0** [comment]
That's interesting.. As long as the `_site/assets/img` folder is preserved, rebuilds should be faster.
I [did change](https://github.com/AleksandrHovhannisyan/aleksandrhovhannisyan.com/pull/99/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) the `clean` script to `rimraf '_site/!(assets)' '_site/assets/!(images)'` to preserve it, could that be not working somehow? Which OS are you on?

Here is what I ran on linux:
```bash
git clone https://github.com/zeroby0/aleksandrhovhannisyan.com.git
git checkout cache
yarn
yarn run build # 42.6 seconds
yarn run build # 10.3 seconds
```

**Comment by AleksandrHovhannisyan** [comment]
Sorry for the delayed response—I'm on Windows (WSL), so that's likely the issue.

The good news is that the local builds aren't _slower_, and if I ever get a better dev setup going, they'll be _better_ 😄

**Comment by zeroby0** [comment]
Maybe, I'll don't have a windows machine around, but I'll try to see it in a VM in the weekend.

Also, the PR is under review and I'm so excited!

**Comment by AleksandrHovhannisyan** [comment]
I noticed! They even tweeted about it 😄

**Comment by zeroby0** [comment]
@AleksandrHovhannisyan 
You're running WSL 2, right?

I tested it directly without wsl, and that worked as expected. Will test wsl now.

**Comment by AleksandrHovhannisyan** [comment]
@zeroby0 I'm actually running WSL1 like a scrub. (My project is in a Windows dir, so WSL1 is technically faster than 2 for me)

**Comment by zeroby0** [comment]
Smart choice!

So I tried wsl1, wsl2, and bare windows on a VM.
On wsl1, with the folder in a Windows dir, 
```bash
time npm run build # No cache
real 2m37.072s
user 1m37.547s
sys 0m35.422s

time npm run build  # Second run, so images are in cache
real 0m56.767s
user 0m21.453s
sys 0m18.406s
```

So I still have no idea why it isn't working.

Can you run this and see if the cache doesn't work there too: https://github.com/zeroby0/demo-eleventy-img-cache

**Comment by AleksandrHovhannisyan** [comment]
Sure! I'll give it a shot when I get a chance.

**Comment by AleksandrHovhannisyan** [comment]
@zeroby0 Wow I'm dumb 😅 I was testing without doing a `yarn install`. Should probably add a `postcheckout` git hook to install dependencies.

18 seconds now! Amazing stuff—will follow the 11ty PR closely and then we can point this PR to the plugin.

**Comment by zeroby0** [comment]
Haha, happens  😄  I do that sometimes too 😂 

TIL about the postcheckout git hook! :D

Can't believe it's been a month since we started this PR, seems like yesterday!

**Comment by AleksandrHovhannisyan** [comment]
@zeroby0 Right? I'm excited to merge this!

When you get a chance, could you update the branch to point to the 11ty image `v1.0.0-beta.0` release? And then I'll test one more time locally + on the deploy preview, and then we should be good.

**Comment by zeroby0** [comment]
Done! :D

But you should probably wait for the second beta before merging if you're using remote URLs: https://github.com/11ty/eleventy-img/issues/123

**Comment by AleksandrHovhannisyan** [comment]
Nope, no remote URLs yet (I may support them in the future though)

**Comment by AleksandrHovhannisyan** [comment]
Works great, thanks again!

**Pull request merged**
